packages/base/src/valibot/freeze.ts

```diff
@@ -0,0 +1,12 @@
+import { transform } from 'valibot';
+
+/**
+ * Valibot pipe action to freeze the array/object. The value will not be cloned before freeze.
+ *
+ * The `readonly()` pipe action in Valibot does not freeze.
+ *
+ * @returns
+ */
+export default function freeze<T>() {
```

Comment:
Migrating to own graph means we can freeze nodes without an issue. We won't touch activities.